pyaedt/edb_core/stackup.py

```diff
@@ -762,3 +763,50 @@ def stackup_limits(self, only_metals=False):
             )
         h_stackup = abs(float(topz) - float(bottomz))
         return topl.GetName(), topz, bottoml.GetName(), bottomz
+
+    @pyaedt_function_handler()
+    def set_etching_layers(self, simulation_setup=None):
+        if not isinstance(simulation_setup, SimulationConfiguration):
+            return False
+        cell = self._builder.cell
+        this_lc = self._edb.Cell.LayerCollection(cell.GetLayout().GetLayerCollection())
+        all_layers = list(this_lc.Layers(self._edb.Cell.LayerTypeSet.AllLayerSet))
+
+        # SignalLayers = list(allLayers.Where(lambda lyr: lyr.GetLayerType() == edb.Cell.LayerType.SignalLayer))
+        signal_layers = [lay for lay in all_layers if lay.GetLayerType() == self._edb.Cell.LayerType.SignalLayer]
+
+        new_layers = list(all_layers.Where(lambda lyr: lyr.GetLayerType() != self._edb.Cell.LayerType.SignalLayer))
+
+        if simulation_setup.signal_layer_etching_instances:
+            for Lay in signal_layers:
```

Comment:
If possible, rename the variable `Lay` to follow snake_case convention.
```suggestion
            for layer in signal_layers:
```